src/daily-dropoffs-report.tsx

```diff
@@ -0,0 +1,392 @@
+  const innerHeight = height - margin.top - margin.bottom;
+
+  const x = scalePoint()
+    .domain(STEPS.map(({ key }) => key))
+    .range([0, innerWidth]);
+
+  const y = scaleLinear()
+    .domain([0, max(data || [], (d) => d.welcome) as number])
+    .range([innerHeight, 0]);
+
+  const line = d3Line()
+    .x((d) => x((d as unknown as StepCount).step) as number)
+    .y((d) => y((d as unknown as StepCount).count) as number) as (s: StepCount[]) => string;
+
+  return (
+    <svg
+      ref={ref}
+      height={height}
+      width={width}
+      onPointerLeave={() => setHighlightedIssuer(undefined)}
+    >
+      <Axis axis={axisLeft(y)} transform={`translate(${margin.left}, ${margin.top})`} />
+      <Axis
+        axis={axisBottom(x).tickFormat(stepToTitle as (s: string) => string)}
+        transform={`translate(${margin.left}, ${margin.top + innerHeight})`}
+        className="x-axis"
+        rotateLabels={width < 700}
+      />
+      <text x={margin.left + innerWidth} y={margin.top} className="title" text-anchor="end">
+        {highlightedRow?.friendly_name}
+      </text>
+      <g transform={`translate(${margin.left}, ${margin.top})`}>
+        {(data || []).map((row) => (
+          <path
+            d={line(toStepCounts(row))}
+            fill="none"
+            stroke={color(row.issuer)}
+            stroke-width="1"
+            onPointerEnter={() => setHighlightedIssuer(row.issuer)}
+          />
+        ))}
+      </g>
+      <g transform={`translate(${margin.left}, ${margin.top})`}>
+        {highlightedRow && (
+          <g className="dots">
+            {toStepCounts(highlightedRow).map(({ step, count, percentOfFirst }, idx) => (
+              <>
+                <circle cx={x(step)} cy={y(count)} r="3" fill={color(highlightedRow.issuer)} />
+                <text x={x(step)} y={y(count)} font-size="12" dx="3" dy="-3">
+                  <tspan x={x(step)}>{formatWithCommas(count)}</tspan>
+                  {idx > 0 && (
+                    <tspan x={x(step)} dy="1.2em">
+                      (${formatAsPercent(percentOfFirst)})
+                    </tspan>
+                  )}
+                </text>
+              </>
+            ))}
+          </g>
+        )}
+      </g>
+    </svg>
+  );
+}
+
+function DailyDropffsReport(): VNode {
+  const ref = useRef(null as HTMLDivElement | null);
+  const [width, setWidth] = useState(undefined as number | undefined);
+  const { setAgencies } = useContext(AgenciesContext);
+  const { start, finish, agency, env } = useContext(ReportFilterContext);
+
+  const { data } = useQuery(`dropoffs/${start.valueOf()}-${finish.valueOf()}`, () =>
+    loadData(start, finish, env)
+  );
+
+  const issuerColor = scaleOrdinal(schemeCategory10);
+
+  useEffect(() => {
+    if (!data) {
+      return;
+    }
+
+    const allAgencies = Array.from(new Set(data.map((d) => d.agency)))
+      .filter((x) => !!x)
+      .sort();
+
+    setAgencies(allAgencies);
+  }, [data]);
```

Comment:
If this is the same as the other report, we could create a custom hook, like `useAgencies(data);`

Reply:
good point... will do

Reply:
got the commit message wrong but extracted in 3bf874d9ded181dedbf4ae54486f364b583ed9c1